snapcraft/internal/lifecycle/_clean.py

```diff
@@ -36,56 +36,51 @@ def _reverse_dependency_tree(config, part_name):
     return dependents
 
 
-def _clean_part_and_all_dependents(part_name, step, config, staged_state,
-                                   primed_state):
-    # Obtain the reverse dependency tree for this part. Make sure all
-    # dependents are cleaned.
-    dependents = _reverse_dependency_tree(config, part_name)
-    dependent_parts = {p for p in config.all_parts
-                       if p.name in dependents}
-    for dependent_part in dependent_parts:
-        dependent_part.clean(staged_state, primed_state, step)
+def _clean_part(part_name, step, config, staged_state, primed_state):
+    if step == common.COMMAND_ORDER[-1]:
```

Comment:
to avoid needing to look at COMMAND_ORDER over and over again, mind adding a quick note to the check?

Reply:
This has been completely reworked thanks to #2159.